### ISSUE TITLE
Fix crash

### ISSUE THREAD

**Issue by Geal** [sozu-proxy/sozu#51]
In gitlab by @Geal on Jun 9, 2016, 16:04

The following command crashes the proxy
```
printf '{"id": "ID_EFGH", "listener": "HTTP", "command":{"type":"ADD_HTTP_FRONT","data":{"app_id":"aa", "hostname": "cltdl.fr","path_begin": ""}}}\0' |socat STDIO UNIX-CONNECT:./command_folder/sock
```

**Comment by Geal** [comment]
In gitlab by @Geal on Jul 8, 2016, 15:38

Fix; use serde, make complete TLS add messages like `{ "id": "ID_42", "type": "PROXY", "listener": "TLS","data":{"type": "ADD_TLS_FRONT", "data": {"app_id": "xxx", "hostname": "yyy", "path_begin": "xxx", "certificate":"<PEM data here>", "key": "<PEM data here>", "certificate_chain": ["<PEM data here>", "<PEM data here>"] }} }`

**Comment by Geal** [comment]
In gitlab by @Geal on Oct 3, 2016, 17:06

is it fixed?

**Comment by Geal** [comment]
In gitlab by @Geal on Oct 4, 2016, 17:42

Fixed in 9c3ec8a
